### ISSUE TITLE
Error in console (Cannot read property 'isContextFor' of null) for list plugin

### ISSUE THREAD

**Issue by luanorlandi** [ckeditor/ckeditor-dev#1439]
## Are you reporting a feature request or a bug?

Bug.

## The issue is already reported:
There is a very similar issue resolved here https://github.com/ckeditor/ckeditor-dev/issues/478.

## Provide detailed reproduction steps (if any)

1. Go to http://cdn.ckeditor.com/4.7.3/full-all/samples/
5. Select part of the string (e.g. "ance of CKEditor.")
6. Click below the text (blinking cursor will disappear)
7. Click "List" buttons (numbered or bullet list)

![ckeditor](https://user-images.githubusercontent.com/6919329/34841210-a08403cc-f6ee-11e7-8d5c-f65338f4218b.gif)

### Expected result

No JavaScript error in console.

### Actual result

JavaScript error in console:

```
ckeditor.js:192 Uncaught TypeError: Cannot read property 'isContextFor' of null
    at window.CKEDITOR.window.CKEDITOR.dom.CKEDITOR.command.refresh (ckeditor.js:192)
    at window.CKEDITOR.window.CKEDITOR.dom.CKEDITOR.command.exec (ckeditor.js:948)
    at window.CKEDITOR.window.CKEDITOR.dom.CKEDITOR.command.exec (ckeditor.js:192)
    at a.execCommand (ckeditor.js:260)
    at CKEDITOR.ui.button.CKEDITOR.tools.extend.click (ckeditor.js:639)
    at Object.execute (ckeditor.js:640)
    at ckeditor.js:641
    at ckeditor.js:30
    at Object.callFunction (ckeditor.js:30)
    at HTMLAnchorElement.onclick ((index):1)
```

## Other details

* Browser: Google Chrome (version 63.0.3239.132 (Official Build) (64-bit))
* CKEditor version: 4.7.3
* Installed CKEditor plugins: Full

**Comment by mlewand** [comment]
I can confirm it. Reproduced with CKE 4.8.0 full, standard, basic preset.

Interestingly there is no selection in the editor in this case.

As someone will have a look on it here's what I see:

* Link has no affect on this issue, it could be just a paragraph with text and it's still reproducible.
* From what I have seen is that you need to have a ranged selection, all the way to the end of a paragraph, and then click below _and after or within_ (left to right-wise) the selection box.

![image](https://user-images.githubusercontent.com/5353898/34852263-09c047bc-f72e-11e7-8cbb-9a6fc0b85d2c.png)

I was able to reproduce it on Windows 10.

On Ubuntu it's even worse, as clicking any area below the paragraph will not put a proper selection.

I'd like to know how it behaves on a MacOS. 

It does look to me like a Chrome upstream issue, that we did not take seriously earlier. We could patch more, and more methods not to fail when there are no selections in the editor, but that'd not be fixing a source of issues.

**Comment by mlewand** [comment]
For the time being I'll set it as confirmed, but after a closer review I expect it will turn to an upstream. Definitely something worth checking.

**Comment by engineering-this** [comment]
After research I've reported an upstream issue in Chromium issue tracker.

https://bugs.chromium.org/p/chromium/issues/detail?id=861673

Chrome is returning selection with 0 ranges when clicking outside of contenteditable, even when contenteditable still has focus.

I'm closing this issue as upstream.

**Comment by jswiderski** [comment]
From what i have checked this problem occurs only in Chrome and only Win 10 and Linux. On Mac it works fine.

**Comment by jswiderski** [comment]
From what I have checked this problem occurs only in Chrome and only in Win 10 and Linux. On Mac it works fine.

During my tests I noticed that selection gets lost whenever user selects end of text e.g. in a paragraph and then clicks on `body` element margin either top, bottom or right (left for some reason worked fine). It was then enough to click on List button to get an error. 

![error](https://user-images.githubusercontent.com/5314687/43832121-a262f916-9b06-11e8-8cbe-741ba4282291.png)


In order to work around this problem I have set `min-height` for `body` element so that it is bigger than the editor content area viewport.
